src/ripple/overlay/impl/PeerSet.cpp

```diff
@@ -73,8 +73,15 @@ bool PeerSet::insert (std::shared_ptr<Peer> const& ptr)
 void PeerSet::setTimer ()
 {
     mTimer.expires_from_now(mTimerInterval);
-    mTimer.async_wait (std::bind (&PeerSet::timerEntry,
-        pmDowncast (), beast::asio::placeholders::error));
+    mTimer.async_wait (
+        [wptr=pmDowncast()](boost::system::error_code const& ec)
+        {
+            if (ec == boost::asio::error::operation_aborted)
+                return;
+
+            if (auto ptr = wptr.lock ())
+                ptr->execute ();
+        });
```

Comment:
@seelabs Do you know about this?
https://github.com/ripple/rippled/blob/develop/src/ripple/beast/utility/weak_fn.h#L119